### PR TITLE
[SPARK-18374][ML]Incorrect words in StopWords/english.txt

### PR DESCRIPTION
**Pull request by hhbyyh**
## What changes were proposed in this pull request?

Currently English stop words list in MLlib contains only the argumented words after removing all the apostrophes, so "wouldn't" become "wouldn" and "t". Yet by default Tokenizer and RegexTokenizer don't split on apostrophes or quotes.

Adding original form to stop words list to match the behavior of Tokenizer and StopwordsRemover. Also remove "won" from list.

see more discussion in the jira: https://issues.apache.org/jira/browse/SPARK-18374

## How was this patch tested?
existing ut

### CODE REVIEW

**Comment by hhbyyh** [comment]
@srowen Thanks for the comments in jira.

**Comment by HyukjinKwon** [comment]
Oh @hhbyyh, I think I am not used to this area but actually I just wanted leave a comment just for stop words list that I considered for my elasticsearch cluster before. I just made a list for missing ones via

```python
target = open("mllib/src/main/resources/org/apache/spark/ml/feature/stopwords/english.txt").read().split("\n")
source = open("mylist.txt").read().split("\n")
for w in source:
    if w not in target:
        print w
```

and I got these below:

```
cannot
could
here's
how's
let's
ought
that's
there's
what's
when's
where's
who's
why's
would
```

I am not sure if they should be added here or not but I just wanted to let you know what I considered before for stop words.

**Comment by srowen** [comment]
Seems reasonable to me too.

**Comment by hhbyyh** [comment]
Thanks @HyukjinKwon. I'll add them.

File: mllib/src/main/resources/org/apache/spark/ml/feature/stopwords/english.txt [hunk 1]
```diff
@@ -149,5 +149,58 @@ shan
 shouldn
 wasn
 weren
-won
 wouldn
```
Comment (srowen) [inline mllib/src/main/resources/org/apache/spark/ml/feature/stopwords/english.txt]: You would then remove the other stems like "wasn" "weren" etc right?

Comment (hhbyyh) [inline mllib/src/main/resources/org/apache/spark/ml/feature/stopwords/english.txt]: I'm fine with both options, leaving them or removing them.

Comment (srowen) [inline mllib/src/main/resources/org/apache/spark/ml/feature/stopwords/english.txt]: I think we should remove the spurious stems; the info on this PR suggests they aren't supposed to be there.

**Review by srowen — APPROVED** [review]

**Comment by hhbyyh** [comment]
Thanks for the review.

**Comment by srowen** [comment]
Merged to master

**Pull request closed without merging**